internal/sbi/producer/data_repository.go

```diff
@@ -1436,19 +1451,28 @@ func PolicyDataUesUeIdSmDataGetProcedure(collName string, ueId string, snssai mo
 ) (*models.SmPolicyData, *models.ProblemDetails) {
 	filter := bson.M{"ueId": ueId}
 
-	if !reflect.DeepEqual(snssai, models.Snssai{}) {
-		filter["smPolicySnssaiData."+util.SnssaiModelsToHex(snssai)] = bson.M{"$exists": true}
-	}
-	if !reflect.DeepEqual(snssai, models.Snssai{}) && dnn != "" {
-		dnnKey := util.EscapeDnn(dnn)
-		filter["smPolicySnssaiData."+util.SnssaiModelsToHex(snssai)+".smPolicyDnnData."+dnnKey] = bson.M{"$exists": true}
-	}
-
-	smPolicyData, pd := getDataFromDB(collName, filter)
+	smPolicyData, pd := getDataFromDBWithArg(collName, filter, 2)
```

Comment:
update util pkg to the latest and replace all `2` to `mongoapi.COLLATION_STRENGTH_IGNORE_CASE`:
```
smPolicyData, pd := getDataFromDBWithArg(collName, filter, mongoapi.COLLATION_STRENGTH_IGNORE_CASE)
```